### DISCUSSION TITLE
Getting "AttributeError: 'Product' object has no attribute 'node'"

### DISCUSSION THREAD

**Discussion by vikrantocode** [saleor/saleor#7915] [Q&A]
I want to add functionality product QA. I start a new app for that and create API to add data but when I try to get product detail it reflects "AttributeError: 'Product' object has no attribute 'node'"

Here is my model

class QA (models.Model):
    product = models.ForeignKey(Product, related_name="product", on_delete=models.CASCADE)
    qa_by = models.ForeignKey(User, on_delete=models.SET_NULL, null=True , blank=True)
    qa_status = models.CharField(max_length=10, choices=STATUS_CHOICE, default="Pending")
    qa_date = models.DateTimeField(null=True , blank=True)

Here is my Mutation

class QAType(CountableDjangoObjectType):
    class Meta:
        interfaces = [graphene.relay.Node]
        model = QA
        fields="__all__"
    
class ProductQAInput(graphene.InputObjectType):
    product = graphene.ID(description="Product ID.", required=True)
    qa_by = graphene.ID(description="Staff User ID.", required=True)
    qa_status = graphene.String(description="QA Status if changed .")
    qa_date = graphene.String(description="QA Date .")

class CreateProductQA(ModelMutation):

    class Arguments:
        input = ProductQAInput(
         description="Fields required"
        )
   
    class Meta:
        description = " QA."
        model = QA
        error_type_class = QAError
        error_type_field = "qa_errors"

    @classmethod
    def perform_mutation(cls, _root, info, **data):
        response = super().perform_mutation(_root, info, **data)
        return response

**Comment by karolkielecki** [comment]
@vikrantocode Hi! Could you please provide more information?

**Comment by vikrantocode** [comment]
@karolkielecki 
I get all information apart from product info, when I try to get product info it returns the same error

{
  "errors": [
    {
      "message": "'Product' object has no attribute 'node'",
      "extensions": {
        "exception": {
          "code": "AttributeError",
          "stacktrace": [
            "Traceback (most recent call last):",
            "  File \"/usr/local/lib/python3.9/site-packages/promise/promise.py\", line 87, in try_catch",
            "    return (handler(*args, **kwargs), None)",
            "  File \"/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py\", line 531, in <lambda>",
            "    lambda resolved: complete_value(",
            "  File \"/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py\", line 546, in complete_value",
            "    return complete_nonnull_value(",
            "  File \"/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py\", line 739, in complete_nonnull_value",
            "    completed = complete_value(",
            "  File \"/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py\", line 571, in complete_value",
            "    return complete_object_value(",
            "  File \"/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py\", line 712, in complete_object_value",
            "    if return_type.is_type_of and not return_type.is_type_of(result, info):",
            "  File \"/app/truefacet/graphql/channel/types.py\", line 36, in is_type_of",
            "    return super().is_type_of(root.node, info)",
            "AttributeError: 'Product' object has no attribute 'node'"
          ]
        }
      }
    }
  ],
  "data": {
    "createProductQA": {
      "product": null
    }
  }
}

**Reply by chikkugadu** [reply]
did you solve it? I am having the same problem. Please let me know when you have a solution. Thanks in advance

**Reply by karolkielecki** [reply]
@vikrantocode Is it Saleor 3.0?

**Reply by vikrantocode** [reply]
@karolkielecki 
no, it is 2.11

**Reply by karolkielecki** [reply]
Would you mind using Saleor 3.0? We recommend our community using 3.0 as it introduces multichannel, improvements, bug fixes, and many more key features. If the issue appears in 3.0, please let me know.

**Reply by vikrantocode** [reply]
@karolkielecki 
I have updated the code according to you but the problem is still the same.
